apps/dc_tools/odc/apps/dc_tools/fs_to_dc.py

```diff
@@ -45,16 +52,22 @@ def cli(input_directory, update_if_exists, allow_unsafe, stac, glob):
     dc = datacube.Datacube()
     doc2ds = Doc2Dataset(dc.index)
 
+    if glob is None:
+        glob = "**/*.json" if stac else "**/*.yaml"
+
     files_to_process = _find_files(input_directory, glob, stac=stac)
 
     added, failed = 0, 0
 
     for in_file in files_to_process:
         with in_file.open() as f:
             try:
-                # Yaml loads as JSON
-                metadata = json.load(f)
-                if stac_transform:
+                if "json" in glob:
```

Comment:
this would be better as `in_file.endswith(".json")` which is a more direct way than to check glob pattern that might contain `json` as part of the path.